### PR TITLE
Restockfix for wood, iron and flour

### PR DESCRIPTION
**Pull request by Krarilotus**
Added the option to modify the restocking interval for important resources that the AIs castle economy uses:
- wood
- iron
- flour

### CODE REVIEW

File: UnofficialCrusaderPatch/Version.cs [hunk 1]
```diff
@@ -624,6 +624,39 @@ public class Version
 
             #region AI LORDS
 
+            new Change("ai_resources_rebuy", ChangeType.AILords, true)
+            {
+                // vanilla:
+                // additional attack troops = factor * attack number
```
Comment (gynt) [inline UnofficialCrusaderPatch/Version.cs:630]: This does not make sense? I don't think resources are related to attack troops?

File: UnofficialCrusaderPatch/Version.cs [hunk 2]
```diff
@@ -624,6 +624,39 @@ public class Version
...
+                    // 004CDEDC
+                    new BinaryEdit("ai_nowood_maxtime")
+                    {
+                        new BinByteValue(),
+                    }
+                },
+
+                new SliderHeader("ai_noiron_maxtime", true, 0, 72, 1, 36, 4)
+                {
+                    // 004CDEDC
```
Comment (gynt) [inline UnofficialCrusaderPatch/Version.cs:643]: Why is this the same address as the one at line 634?

File: UnofficialCrusaderPatch/Version.cs [hunk 3]
```diff
@@ -624,6 +624,39 @@ public class Version
...
+                    new BinaryEdit("ai_noiron_maxtime")
+                    {
+                        new BinByteValue(),
+                    }
+                },
+
+                new SliderHeader("ai_noflour_maxtime", true, 0, 72, 1, 36, 6)
+                {
+                    // 004CDEDC
```
Comment (gynt) [inline UnofficialCrusaderPatch/Version.cs:652]: Same here.

File: UnofficialCrusaderPatch/Version.cs [hunk 4]
```diff
@@ -624,6 +624,39 @@ public class Version
 
             #region AI LORDS
 
+            new Change("ai_resources_rebuy", ChangeType.AILords, true)
+            {
+                // vanilla:
+                // additional attack troops = factor * attack number
+
+                new SliderHeader("ai_nowood_maxtime", true, 0, 72, 1, 36, 1)
```
Comment (gynt) [inline UnofficialCrusaderPatch/Version.cs:632]: Please document what this feature does. What does the original value of 36 (0x24) mean and why should its suggested value be 1?

This feature is on by default and set to a very much lower number than the original. This needs a good testing before we release it, it should not cripple any AI.

**Review by gynt — COMMENTED** [review]
I put some thoughts and remarks in comments.

File: UnofficialCrusaderPatch/Localization/English.txt [hunk 5]
```diff
@@ -2,6 +2,31 @@
...
 // ======================
 
+ai_resources_rebuy
+{
+"Improve AI restocking of needed goods"
+}
+
+ai_resources_rebuy_descr
+{
+"The AI often doesn't restock goods even if it has the gold for it, leaving its production chains running dry. This option reduces the amount of time the AI waits for its stocks to be empty before restocking the necessary goods."
```
Comment (gynt) [inline UnofficialCrusaderPatch/Localization/English.txt:12]: The description is really good, but does not show up in the GUI, it is needed for context though.

I rephrased it a tiny bit to this:
The AI often doesn't restock goods even if it has the gold for it, leaving its production chains running dry. This option reduces the amount of time the AI waits for stocks to refill before restocking the necessary goods by buying.

File: UnofficialCrusaderPatch/Localization/English.txt [hunk 6]
```diff
@@ -2,6 +2,31 @@
...
+}
+
+ai_nowood_maxtime_descr
+{
+"Time in 1/12th of months until the AI restocks 5 wood."
```
Comment (gynt) [inline UnofficialCrusaderPatch/Localization/English.txt:17]: If we can't have the long description, then maybe this short description should be a little more descriptive:
"The AI waits for empty stocks to refill by themselves before buying the goods in the market. Set the time in 1/12th of months until the AI buys 5 wood."

**Review by gynt — COMMENTED** [review]
Some comments on the descriptions. Shame we can't have the long description in SliderHeaders. Is this known @patel-nikhil  ?

**Comment by Krarilotus** [comment]
> If we can't have the long description, then maybe this short description should be a little more descriptive:
"The AI waits for empty stocks to refill by themselves before buying the goods in the market. Set the time in 1/12th of months until the AI buys 5 wood."

@gynt unfortuantely that is not really an option, as it would make this option span over multiple lines. all other options short descriptions are designed to neatly fit into one single line inside the patch. Thats what short and long descriptions are for.

**Comment by gynt** [comment]
@Krarilotus but the one right beneath: "Let the AI build houses smarter" also has multiline descriptions, right? Why is it a problem?

**Comment by Krarilotus** [comment]
ahh i missunderstood what you ment. I thought you ment the short description. But you meant the slider description instead. Yeah we could verbose that i suppose, i am open for suggestions. We need to look over all the new features texts anyways before we go into translation again next week.

**Pull request merged**